src/frontend/org/voltcore/zk/SynchronizedStatesManager.java

```
@@ -535,10 +607,26 @@ private void checkForBarrierParticipantsChange() {
                                 }
                                 unlockLocalState();
                                 if (type == REQUEST_TYPE.STATE_CHANGE_REQUEST) {
-                                    stateChangeProposed(proposedState);
+                                    try {
+                                        stateChangeProposed(proposedState);
+                                    } catch (Exception e) {
+                                        if (m_log.isDebugEnabled()) {
+                                            m_log.debug("Error in StateMachineInstance callbacks.", e);
+                                        }
+                                        m_deregistered = true;
+                                        m_shared_es.submit(new CallbackExceptionHandler(this));
+                                    }
                                 }
                                 else {
-                                    taskRequested(proposedState);
+                                    try {
+                                        taskRequested(proposedState);
```

Perhaps you should wrap this in a catch block. I am not sure whether you want to fail outright at this point or ignore the exception and continue as if nothing has happened. In either case I don't think you want the executor to catch this exception.